python/feldera/rest/feldera_client.py

```diff
@@ -342,6 +342,39 @@ def shutdown_pipeline(self, pipeline_name: str, timeout_s: Optional[float] = 300
             f"timeout error: pipeline '{pipeline_name}' did not shutdown in {timeout_s} seconds"
         )
 
+    def suspend_pipeline(self, pipeline_name: str, timeout_s: Optional[float] = 300):
+        """
+        Suspend a pipeline
+
+        :param pipeline_name: The name of the pipeline to suspend
+        :param timeout_s: The amount of time in seconds to wait for the pipeline to suspend. Default is 300 seconds.
+        """
+
+        if timeout_s is None:
+            timeout_s = 300
+
+        self.http.post(
+            path=f"/pipelines/{pipeline_name}/suspend",
+        )
+
+        start = time.monotonic()
+
+        while time.monotonic() - start < timeout_s:
+            status = self.get_pipeline(pipeline_name).deployment_status
+
+            if status == "Suspended":

```

Comment:
Another possible outcome is status == "Failed", in which case this function should return an error.